src/pins/gen.rs

```diff
@@ -97,13 +106,26 @@ macro_rules! pins {
             }
         }
 
-        /// A token representing a pin
+        /// A token representing a pin.
+        /// Tokens are unique for each pin, and ensure that no pins are re-used by accident.
+        /// They also hold the `pin::Trait` information for their associated pin.
```

Comment:
This isn't quite correct. Pin types not being `Copy`/`Clone` and having private constructors ensure that pins are unique and not re-used by accident. `Token` is a somewhat convoluted way to ensure the user doesn't disable the GPIO peripheral after having initialized GPIO pins. This is documented in `GPIO`, I think.

```suggestion
```